apostrophecms/apostrophe
# fix 404 error when trying to edit a draft with a different slug

## Summary

Fix 404 error when trying to view or edit a document, when its draft was saved with a different slug than the published version and its page -or piece show page- has never been visited before:

![image](https://user-images.githubusercontent.com/8301962/229494173-9ec207ed-ccb0-4d1a-aee9-520652c7815e.png)

Rely on the doc's `aposDocId` to redirect to the URL of the draft version as a fallback.

## What are the specific steps to test this change?

In testbed (or any A3 project) with `article` pieces:

- create an article and give it a title: "one"
- save it and publish it
- now edit it and change the title: "first" (this is just an example, the important thing is that the slug changes)
- save the draft
- go to the article show page using the published version of its slug, or:
  - go to the article index page (create it if not already existing)
  - go to the article "one" (in the published version of the article index page)
![image](https://user-images.githubusercontent.com/8301962/229494306-7a5bc981-5ab2-420b-a0ef-a00d3c93e806.png)
- in the article's show page hit the "Edit" button
![image](https://user-images.githubusercontent.com/8301962/229494429-273bf6bb-b7a0-4c7e-9414-c448b8961121.png)

The following error message should not appear:

![image](https://user-images.githubusercontent.com/8301962/229494364-ba69b594-c3a6-4c45-a4a1-f6218f042a0e.png)

There should be a redirection to the draft doc URL.

Same goes for pages.

## What kind of change does this PR introduce?
*(Check at least one)*


## Make sure the PR fulfills these requirements:

- [x] Related tests have been updated --> https://github.com/apostrophecms/testbed/pull/169

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**
